index.js

```diff
@@ -8,35 +8,44 @@
 (function () {
 	'use strict';
 
-	var hasOwn = {}.hasOwnProperty;
-
-	function classNames () {
-		var classes = [];
-
-		for (var i = 0; i < arguments.length; i++) {
-			var arg = arguments[i];
-			if (!arg) continue;
-
-			var argType = typeof arg;
-
-			if (argType === 'string' || argType === 'number') {
-				classes.push(arg);
-			} else if (Array.isArray(arg) && arg.length) {
-				var inner = classNames.apply(null, arg);
-				if (inner) {
-					classes.push(inner);
+	var hasOwnProperty = Object.prototype.hasOwnProperty;
+	var isArray = Array.isArray;
+
+	function reduceArray (args) {
+		var len = args.length;
+		if (!len)
+			return "";
+		var str = "", item, type, i, n;
+		for (i = 0; i < len; i++) {
+			if (!(item = args[i]))
+				continue;
+			type = typeof item;
+			if (type === "string" || type === "number") {
```

Comment:
The only thing which i dont get - why do we have to pass the numbers here?
Due to [W3C's TR](http://www.w3.org/TR/CSS21/grammar.html#scanner) - they're NOT valid class names.

Reply:
This handles use cases like this:

```js
classNames('foo', 23); // => 'foo 23'
```

Reply:
I know what it does, 23 is invalid classname (as any number leaded string will be), that's I wonder why this lib passes it to result.